src/component/Restaurants.js

```diff
@@ -1,35 +1,57 @@
 import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { RestaurantListItem } from "./RestaurantListItem";
 
-const api = "/cities";
+import thumbnail1Src from "place-my-order-assets/images/1-thumbnail.jpg";
+import thumbnail2Src from "place-my-order-assets/images/2-thumbnail.jpg";
+import thumbnail3Src from "place-my-order-assets/images/3-thumbnail.jpg";
+import thumbnail4Src from "place-my-order-assets/images/4-thumbnail.jpg";
 
 /**
  * Displays information about available restaurants.
  * @returns {JSX.Element}
  */
 function Restaurants (props) {
-    const [region, setRegion] = useState("");
     const [cities, setCities] = useState({});
-    const [city, setCity] = useState("");
+    const [city, setCity] = useState(DEFAULT_SELECT_VALUE);
+    const [region, setRegion] = useState(DEFAULT_SELECT_VALUE);
+    /** @type {[RestaurantRegionData, (value: RestaurantRegionData) => void]} */
+    const [restaurants, setRestaurants] = useState({});
 
+    function cityChange(evt) {
+        const { value: nextCityValue } = evt.target;
+        setCity(nextCityValue);
 
-    const cityChange = evt => {
-        setCity(evt.target.value);
-    };
+        if (nextCityValue && (!restaurants[region] || !restaurants[region][nextCityValue])) {
+            fetchRestaurants(region, nextCityValue);
+        }
+    }
 
-    const regionChange = evt => {
+    function regionChange(evt) {
         const { value: nextRegionValue } = evt.target;
         setRegion(nextRegionValue);
-        setCity("");
+        setCity(DEFAULT_SELECT_VALUE);
 
-        if (!cities[nextRegionValue]) {
+        if (nextRegionValue && !cities[nextRegionValue]) {
             fetchCities(nextRegionValue);
         }
-    };
+    }
 
-    const fetchCities = async cityRegion => {
-        const response = await fetch(`${api}?filter[state]=${cityRegion}`);
-        const body = await response.json();
+    async function fetchCities(cityRegion) {
```

Comment:
This kind of `async` fetching would be considered a "side-effect", and should be isolated in a `useEffect()` hook.

The whole handling of this state, and the fetching of data, could probably be broken out into a custom hook here (that uses useEffect, and maybe useReducer or something), or maybe not, but it certainly needs to isolate the side-effects into the `useEffect` hook

Reply:
I think the custom hook can be useful, but I'm not sure this is complex enough to justify the cognitive load.

Reply:
Moved into useEffect.